lib/fluent/plugin/filter.rb

```diff
@@ -28,24 +28,57 @@ class Filter < Base
       include PluginLoggerMixin
       include PluginHelper::Mixin
 
+      class DuplicatedFilterMethodError < StandardError; end
+
       helpers :event_emitter
 
       def filter(tag, time, record)
         raise NotImplementedError, "BUG: filter plugins MUST implement this method"
       end
 
+      def filter_with_time(tag, time, record)
+        raise NotImplementedError, "BUG: filter plugins MUST implement this method"
+      end
+
       def filter_stream(tag, es)
         new_es = MultiEventStream.new
         es.each do |time, record|
           begin
-            filtered_record = filter(tag, time, record)
-            new_es.add(time, filtered_record) if filtered_record
+            filtered_record = do_filter(tag, time, record)
+            new_es.add(*filtered_record) if filtered_record
           rescue => e
             router.emit_error_event(tag, time, record, e)
           end
         end
         new_es
       end
+
+      private
+
+      def do_filter(tag, time, record)
```

Comment:
Thanks for your comment. I didn't mind such point.
I fixed.
https://github.com/fluent/fluentd/pull/1140/files#diff-3bdc55bd92825d185fcb6fc41334cef3R48